crates/miden-objects/src/note/inputs.rs

```diff
@@ -66,16 +66,13 @@ impl NoteInputs {
 
     /// Returns the note's input formatted to be used with the advice map.
     ///
-    /// The format is `input_len || INPUTS || PADDING`, where:
+    /// The format is `INPUTS || PADDING`, where:
     ///
-    /// - input_len is the number of inputs
+    /// Where:
     /// - INPUTS is the variable inputs for the note
     /// - PADDING is the optional padding to align the data with a 2WORD boundary
     pub fn format_for_advice(&self) -> Vec<Felt> {
-        // NOTE: keep map in sync with the `note::get_inputs` API procedure
-        let mut padded = pad_inputs(&self.values);
-        padded.insert(0, self.num_values().into());
-        padded
+        pad_inputs(&self.values)
     }
```

Comment:
Not for this PR, but currently the logic of how a given objects is injected into the advice input can be located in a number of places (e.g., here and also in `miden-lib`). And it is not easy to tell why certain logic "lives" together with the type (like here) or why it "lives" in the transaction advice input injector module of `miden-lib`. I wonder if there is a good way to come up with a clear methodology for this.

Reply:
I agree. I think for me, the clearest way would be if everything was contained within `miden-lib` as part of `extend_advice_inputs`.
The other approach is to move it to the individual types. That doesn't make much sense to me, because this format is specific to the transaction kernel / advice provider so it being "closer" to that, i.e. in miden-lib, seems best. Also, in the future, we might have multiple ways to add a given type to the advice provider when we introduce the batch or block kernels.